src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -1,32 +1,122 @@
+            _fromCacheOption = parseResult.ValueForOption<string>(WorkloadInstallCommandParser.FromCacheOption);
         }
 
         public override int Execute()
         {
             // TODO stub
-            Reporter.Output.WriteLine($"WIP workload install {string.Join("; ",_workloadIds)}");
+            Reporter.Output.WriteLine($"WIP workload install {string.Join("; ", _workloadIds)}");
+            List<string> allowedMockWorkloads = new List<string> {"mobile-ios", "mobile-android"};
+
+            if (!_printDownloadLinkOnly && string.IsNullOrWhiteSpace(_fromCacheOption))
+            {
+                Reporter.Output.WriteLine(
+                    "mock currently does not support normal install. Need --print-download-link-only or --from-cache");
+            }
+
+            if (_workloadIds.Except(allowedMockWorkloads).Any())
+            {
+                Reporter.Output.WriteLine("Only support \"mobile-ios\", \"mobile-android\", \"mobile\" in the mock");
+            }
+
+            SourceRepository source =
+                Repository.Factory.GetCoreV3("https://www.myget.org/F/mockworkloadfeed/api/v3/index.json");
+            ServiceIndexResourceV3 serviceIndexResource = source.GetResourceAsync<ServiceIndexResourceV3>().Result;
+            IReadOnlyList<Uri> packageBaseAddress =
+                serviceIndexResource?.GetServiceEntryUris(ServiceTypes.PackageBaseAddress);
+            List<string> allPackageUrl = new List<string>();
+
+            if (_printDownloadLinkOnly)
+            {
+                if (_workloadIds.Contains("mobile-ios"))
+                {
+                    allPackageUrl.Add(nupkgUrl(packageBaseAddress.First().ToString(), "Microsoft.iOS.Bundle",
+                        NuGetVersion.Parse("6.0.100")));
+
+                    AddNewtonsoftJson(allPackageUrl);
+                }
+
+                if (_workloadIds.Contains("mobile-android"))
+                {
+                    allPackageUrl.Add(nupkgUrl(packageBaseAddress.First().ToString(), "Microsoft.NET.Workload.Android",
+                        NuGetVersion.Parse("6.0.100")));
+
+
+                    AddNewtonsoftJson(allPackageUrl);
+                }
+
+                Reporter.Output.WriteLine("==allPackageLinksJsonOutputStart==");
+                Reporter.Output.WriteLine(JsonSerializer.Serialize(allPackageUrl));
+                Reporter.Output.WriteLine("==allPackageLinksJsonOutputEnd==");
+            }
+
+            if (!string.IsNullOrWhiteSpace(_fromCacheOption))
+            {
+                Directory.CreateDirectory(MockInstallDirectory);
+                if (_workloadIds.Contains("mobile-android"))
+                {
+                    File.Copy(Path.Combine(_fromCacheOption, "Microsoft.NET.Workload.Android.6.0.100.nupkg"),
+                        Path.Combine(MockInstallDirectory, "Microsoft.NET.Workload.Android.6.0.100.nupkg"));
```

Comment:
I'm confused by this section, so this is mocking a workload in which the only pack required is this nupkg? It's enough to just copy with nupkg without extracting it?

Reply:
I just did the minimal to make the API from the commandline look right. I could use an empty file to mark a package is installed or not. Don't read too much about the mock implementation.